### PR TITLE
Create full note edit and speech-to-text functionality

### PR DESCRIPTION
**Pull request by hauptdigital**
In this pull request, I added the functionality to edit note title and content. Here is a more detailed description:

When you are not recording, you will see:
- Input field to enter title
- Textarea to enter or edit your note

When you are recording, you will see:
- Input field to enter title
- A block of text (`div`) that displays the content of the note. (`div` is chosen as a way to make the note content read only while recording).

Functionality in the background is:
- Content of the note is synced between the 'textarea' (`NoteContent` component) and the `div` (`NoteContentReadOnly` component).
- Text added via recording is styled, so that new text chunks will fade in from the top and old text is displayed as gray.

Here is my deployment:
https://dev.deepspeech-notes.haupt.digital/

Title

Note

recording

### CODE REVIEW

File: client/src/components/NoteContent.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  padding: 1rem;
+  width: 100%;
+  max-width: 600px;
+  flex-grow: 2;
+  &:focus {
+    box-shadow: 0 0 0 2px ${(props) => props.theme.colors.altTwo};
+  }
+  &:hover {
+    box-shadow: 0 0 0 2px ${(props) => props.theme.colors.altTwo};
+  }
```
Comment (lmachens) [inline client/src/components/NoteContent.js:23]:
```suggestion
  &:focus, &:hover {
    box-shadow: 0 0 0 2px ${(props) => props.theme.colors.altTwo};
  }
```

File: client/src/components/NoteContentReadOnly.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      {props.noteContent.recognizedText !== '' && (
+        <CSSTransitionGroup
+          transitionName="fadeIn"
+          transitionAppear={true}
+          transitionLeave={false}
+          transitionEnterTimeout={300}
+          transitionAppearTimeout={300}
+        >
+          <TextChunk
+            key={props.noteContent.text + props.noteContent.recognizedText}
```
Comment (lmachens) [inline client/src/components/NoteContentReadOnly.js:46]: Are you sure you need `key` here?

File: client/src/GlobalStyles.js [hunk 3]
```diff
@@ -17,6 +17,23 @@ function GlobalStyle() {
...
+          bottom: 20px;
+        }
+
+        .fadeIn-enter-active,
+        .fadeIn-appear-active {
+          opacity: 1;
+          transition: all 300ms ease-in;
+          position: relative;
+          bottom: 0px;
+        }
```
Comment (lmachens) [inline client/src/GlobalStyles.js:36]: It doesn't look like that this is the correct place for the fade animations. 
I think you need them where you are using react-transition-groups. You could either add another GlobalStyles to the place where you are using the transitions or you take a look at this:

https://stackoverflow.com/questions/42660907/using-reactcsstransitiongroup-with-styled-component

File: client/src/components/NoteTitle.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import styled from '@emotion/styled';
+
+const NoteTitle = styled.input`
```
Comment (lmachens) [inline client/src/components/NoteTitle.js:3]: The styling is very similar to `NoteContent`. 
This is a good case for https://emotion.sh/docs/composition or https://emotion.sh/docs/styled#styling-any-component

File: client/src/GlobalStyles.js (hunk 3 above)
Comment (hauptdigital) [reply client/src/GlobalStyles.js:36]: I moved the styles into the styled component.

File: client/src/components/NoteContentReadOnly.js (hunk 2 above)
Comment (hauptdigital) [reply client/src/components/NoteContentReadOnly.js:46]: http://reactcommunity.org/react-transition-group/transition-group#TransitionGroup-prop-children

I understood that the `CSSTransitionGroup` component from the module is triggering transition animations if the `key` changes. So I used this behavior to trigger my transition animations.

File: client/src/components/NoteTitle.js (hunk 4 above)
Comment (hauptdigital) [reply client/src/components/NoteTitle.js:4]: I used composition and it works. Thanks!

**Pull request merged**